api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +118,54 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestEnvironmentDeleteDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
```

Comment:
there is too much unused setup and assertions in this test; anything this isn't directly related to testing the specific behavior should be taken out. Also, we should test for each dependent entity type: loadbalancer, service, and task.